staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1/event_expansion.go

```diff
@@ -96,3 +104,53 @@ func (e *events) PatchWithEventNamespace(event *v1beta1.Event, data []byte) (*v1
 		Into(result)
 	return result, err
 }
+
+// Search finds events about the specified object. The namespace of the
+// object must match this event's client namespace unless the event client
+// was made with the "" namespace.
+func (e *events) Search(scheme *runtime.Scheme, objOrRef runtime.Object) (*v1beta1.EventList, error) {
```

Comment:
rather than plumbing scheme and object through this method, a helper function in `pkg/apis/events/helper` that takes an ObjectReference and an event client seems like it would be simpler:

```
func SearchEvents(client EventsV1beta1Interface, ref *v1.ObjectReference) (*v1beta1.EventList, error)
```

callers could get an object reference however they wanted (calling `ref.GetReference` if desired, or via some other method)

Reply:
That could work too

Reply:
@yastij @liggitt May I ask in which directory it would be better to add these helper functions? Thanks a lot.

Reply:
alongside the client-go events tools, maybe? `k8s.io/client-go/tools/events`?

Reply:
sounds reasonable

Reply:
Thanks a lot. :)